codility_training/lessons.lessons08.Leader.EquiLeader/sAp00n.py

```diff
@@ -10,58 +10,27 @@ def solution(A):
 
     else:
         eq_leadernum = 0
-        ind_list = []
         n = len(A)
-        for ind in range(n):
-            if A[ind] == leader:
-                ind_list.append(ind)
-        # print(ind_list)
-        group_dic = {}
-        for i in range(len(ind_list) - 1):
-            # print(f'left ={A[:ind_list[i]+1]},{ind_list[:i+1]}   right = {A[ind_list[i]+1:]}, {ind_list[i+1:]}')
-
-            # case 01
-            # -->
-            left_A = A[:ind_list[i] + 1]
-            left_leader_ind_list = ind_list[:i + 1]
-            right_A = A[ind_list[i] + 1:]
-            right_leader_ind_list = ind_list[i + 1:]
-
-            '''print(f'\ncase 01 -->\n left: {left_A}  right: {right_A}')
-            print(f'indl_left:{left_leader_ind_list}  indl_right:{right_leader_ind_list}')'''
-            dic_key = (str(left_A), str(right_A))
-            if dic_key not in group_dic:
-                group_dic[dic_key] = 0
-                left_condition = len(left_leader_ind_list) / (len(left_A))
-                right_condition = len(right_leader_ind_list) / len(right_A)
-                '''print(f'left_c = {len(ind_list[:i+1])}/{(ind_list[i]+1)} = {left_condition}  right_c = {len(ind_list[i+1:])}/{len(A[ind_list[i]+1:])} = {right_condition}')'''
-                if left_condition and right_condition > 1 / 2:
+        logical_list = [0 for i in range(n)]
+        temp = 0
+        for value in range(n):
+            if A[value] == leader:
+                temp += 1
+            logical_list[value] = temp
+        # print(logical_list)
+        for i in range(len(A) - 1):
+            left_ele_num = len(A[:i + 1])
+            right_ele_num = len(A[i + 1:])
```

Comment:
```suggestion
        left_ele_num = 1
        right_ele_num = len(A) - 1
        for i in range(len(A) - 1):
            left_ele_num += 1
            right_ele_num -= 1
```
이 부분은 반복문 밖에서 카운트 하는건 어떨까요?
아마도 n ** 2 가 나오는 원인은 이 부분인 것 같네요~